src/storage/v2/replication/replication_persistence_helper.cpp

```diff
@@ -70,6 +75,13 @@ std::optional<ReplicaStatus> JSONToReplicaStatus(nlohmann::json &&data) {
       data.at(kSSLKeyFile).get_to(replica_status.ssl->key_file);
       data.at(kSSLCertFile).get_to(replica_status.ssl->cert_file);
     }
+
+    if (data.find(kReplicationRole) != data.end()) {
+      auto role = replication::ReplicationRole::MAIN;  // initializing it to dummy value because of clang-tidy
+      data.at(kReplicationRole).get_to(role);          // here happens the assigning of the role
```

Comment:
replica_status.role = replication::ReplicationRole::MAIN;
data.at(kReplicationRole).get_to(replica_status.role.value());   

https://en.cppreference.com/w/cpp/utility/optional/value

I think this should work. `value()` returns reference if there is value inside, and get_to will store value on that reference, so all good, right?